src/dialogs/quick-open/ha-quick-open-dialog.ts

```diff
@@ -0,0 +1,220 @@
+    }
+    this._items = this._filterItems(items, this._entityFilter);
+  }
+
+  private _generateCommandItems(_filter: string): QuickOpenItem[] {
+    const reloadableDomains = componentsWithService(this.hass, "reload").sort();
+
+    return reloadableDomains.map((domain) => ({
+      text:
+        this.hass.localize(`ui.dialogs.quick_open.commands.${domain}`) ||
+        this.hass.localize(
+          "ui.dialogs.quick_open.commands.reload",
+          "domain",
+          domainToName(this.hass.localize, domain)
+        ),
+      domain,
+      onClick: this._executeCommand,
+    }));
+  }
+
+  private _generateEntityItems(
+    hass: HomeAssistant,
+    _filter: string
+  ): QuickOpenItem[] {
+    return Object.keys(hass.states)
+      .map((key) => hass.states[key])
+      .map<QuickOpenItem>((entity) => ({
+        text: entity.entity_id,
+        domain: computeDomain(entity.entity_id),
+        onClick: this._entityMoreInfo,
+      }));
+  }
+
+  private _filterItems(
+    items: QuickOpenItem[],
+    filter: string
+  ): QuickOpenItem[] {
+    return items
+      .filter((value) => fuzzySequentialMatch(filter.toLowerCase(), value.text))
+      .sort((itemA, itemB) => {
+        if (itemA.text < itemB.text) {
+          return -1;
+        }
+        if (itemA.text > itemB.text) {
+          return 1;
+        }
+        return 0;
+      });
+  }
+
+  private async _executeCommand(ev: Event) {
+    const eventData = {
+      domain: (ev.target as any).domain,
+      service: "reload",
+    };
+
+    await this.hass.callService(eventData.domain, "reload");
+
+    fireEvent(this, "hass-service-called", eventData);
+
+    this.closeDialog();
+  }
+
+  private _entityMoreInfo(ev: Event) {
+    ev.preventDefault();
+    fireEvent(this, "hass-more-info", {
+      entityId: (ev.target as any).text,
+    });
+    this.closeDialog();
+  }
+
+  protected render() {
+    if (!this._opened) {
+      return html``;
+    }
+
+    return html`
+      <ha-dialog open @closed=${this.closeDialog} hideActions>
+        <paper-input
+          dialogInitialFocus
+          no-label-float
+          @value-changed=${this._entityFilterChanged}
+          .label=${this.hass.localize(
+            "ui.dialogs.quick_open.filter_placeholder"
+          )}
+          type="search"
+          value=${this._commandMode
+            ? `>${this._entityFilter}`
+            : this._entityFilter}
+        ></paper-input>
+        <mwc-list>
+          ${this._items.map(
+            ({ text, domain, onClick }) => html`
+              <mwc-list-item
+                @click=${onClick}
+                .domain=${domain}
+                alt=${this.hass.localize(
+                  "ui.panel.developer-tools.tabs.states.more_info"
+                )}
+                .title=${this.hass.localize(
+                  "ui.panel.developer-tools.tabs.states.more_info"
+                )}
+              >
+                <ha-icon .icon=${domainIcon(domain)}></ha-icon>
```

Comment:
https://github.com/material-components/material-components-web-components/tree/master/packages/list#leading-graphic

Make this a graphic item.

Reply:
Had to install mwc-icon (we only have mwc-icon-button atm).

With this code:

```
<mwc-list-item graphic="icon" //...>
  <mwc-icon slot="graphic">
    <ha-icon .icon=${domainIcon(domain)}></ha-icon>
  </mwc-icon>
</mwc-list-item>
```
I get this result:

![image](https://user-images.githubusercontent.com/1480827/95137664-6d0dd080-071d-11eb-90b1-3b0754d211a2.png)

You like?

Reply:
Dont install mwc icon. Just use mwc icon button

Reply:
But yes this looks VERY nice 🎉

Reply:
Actually, dont use mwc icon button

Reply:
just use ha-icon. add slot="graphic" to that

Reply:
Will do. Do I need to keep the `graphic="icon"` bit in `mwc-list-item` or can that be removed?

Reply:
Done and pushed

Reply:
Updated GIF in PR as well.